### PR TITLE
fix: redirect to 404 from subscribe page when api is unpublished.

### PR DESCRIPTION
**Pull request by kovaren**
## Issue

https://gravitee.atlassian.net/browse/APIM-13287

## Description

Fix a case when an API is unpublished and the subscribe page is accessed by URL and appears empty.

https://github.com/user-attachments/assets/137a4b61-611c-4f33-814a-ccc2b3c69a06

### CODE REVIEW

File: gravitee-apim-portal-webui-next/src/resolvers/api.resolver.spec.ts
Comment (aqibmohammadkhan) [inline gravitee-apim-portal-webui-next/src/resolvers/api.resolver.spec.ts:1]: a 404 maps to router.navigate(['/404']) and returns EMPTY. The unit test subscribes directly to the resolver observable and asserts completion, but does not verify what Angular Router does when a resolver completes without emitting during an active navigation.
Risk: depending on router internals/version, this can produce a canceled/errored current navigation rather than a clean redirect, and the current test would still pass.

Could we add a router-level test (via RouterTestingModule + real navigation) to validate that returning EMPTY from the resolver during a 404 actually redirects to /404 and doesn’t leave navigation canceled? Current test only checks observable completion.

File: gravitee-apim-portal-webui-next/src/resolvers/api.resolver.ts [hunk 1]
```diff
@@ -24,4 +25,14 @@ export const apiResolver = ((
   route: ActivatedRouteSnapshot,
   _: RouterStateSnapshot,
   apiService: ApiService = inject(ApiService),
-): Observable<Api> => apiService.details(route.params['apiId'])) satisfies ResolveFn<Api>;
+  router: Router = inject(Router),
+): Observable<Api> =>
+  apiService.details(route.params['apiId']).pipe(
+    catchError((err: unknown) => {
+      if (err instanceof HttpErrorResponse && err.status === 404) {
+        void router.navigate(['/404']);
```
Comment (aqibmohammadkhan) [inline gravitee-apim-portal-webui-next/src/resolvers/api.resolver.ts:33]: Can we use a centralized route constant/helper for the not-found path instead of hardcoding '/404' to avoid route drift?

here we are using a literal path. If route config changes, this can silently drift.

Comment (kovaren) [reply gravitee-apim-portal-webui-next/src/resolvers/api.resolver.ts:33]: This suggestion is valid but there are many other places in the code with a hardcoded 404.

WDYT about handling this in a separate PR to avoid unrelated changes in this PR? I can create a ticket for that.

File: gravitee-apim-portal-webui-next/src/resolvers/api.resolver.spec.ts
Comment (kovaren) [reply gravitee-apim-portal-webui-next/src/resolvers/api.resolver.spec.ts:1]: Fixed

**Review by aqibmohammadkhan — APPROVED** [review]
